### ISSUE TITLE
[ECS Provider] Traefik can't auto-detect region on Fargate

### ISSUE THREAD

**Issue by leshik** [containous/traefik#7196]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

According to #7145, Traefik should take into account the `AWS_REGION` environment variable exposed by AWS Fargate. It seems that's not the case.

### What did you expect to see?

Traefik should auto-discover ECS services of the same region it was launched in.

### What did you see instead?

```
time="2020-08-19T10:48:11Z" level=error msg="Provider connection error EC2MetadataRequestError: failed to get EC2 instance identity document\ncaused by: RequestError: send request failed\ncaused by: Get \"http://169.254.169.254/latest/dynamic/instance-identity/document\": dial tcp 169.254.169.254:80: connect: invalid argument, retrying in 8.54594966s" providerName=ecs
time="2020-08-19T10:48:11Z" level=info msg="No EC2 region provided, querying instance metadata endpoint..." providerName=ecs
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
v2.3.0-rc3
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

AWS Fargate, just the test configuration via CLI options.

```
--api.insecure=true
--log
--log.level=INFO
--ping=true
--providers.ecs.autodiscoverclusters=true
```

IAM policy used exactly matches one from the documentation. Setting region manually with e.g. `--providers.ecs.region=ap-southeast-2` makes autodiscovery work and errors disappear.

**Comment by rtribotte** [comment]
duplicate of #7096

**Comment by leshik** [comment]
Hey @rtribotte, #7096 is closed by #7145, but #7145 doesn’t resolve the issue.

**Comment by rtribotte** [comment]
Hello @leshik,

The fix bring by #7145 will come with the v2.3.0-rc4 that will be released soon, so the region autodiscovery for Fargate will be fixed with this new version.

**Comment by leshik** [comment]
Thanks, it wasn't obvious to me that it's not in `master`.
